core/src/main/java/org/apache/gravitino/listener/api/event/policy/AlterPolicyPreEvent.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.listener.api.event.policy;
+
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.annotation.DeveloperApi;
+import org.apache.gravitino.listener.api.event.OperationType;
+import org.apache.gravitino.policy.PolicyChange;
+
+/** Represents an event that is triggered before altering a policy. */
+@DeveloperApi
+public final class AlterPolicyPreEvent extends PolicyPreEvent {
+  private final PolicyChange[] policyChanges;
+
+  /**
+   * Constructs an instance of {@code AlterPolicyPreEvent}.
+   *
+   * @param user The username of the individual who initiated the alter policy operation.
+   * @param identifier The identifier of the policy to be altered.
+   * @param policyChanges The changes to be applied to the policy.
+   */
+  public AlterPolicyPreEvent(String user, NameIdentifier identifier, PolicyChange[] policyChanges) {
+    super(user, identifier);
+    this.policyChanges = policyChanges != null ? policyChanges.clone() : null;
+  }
+
+  /**
+   * Returns the changes to be applied to the policy.
+   *
+   * @return the policy changes.
+   */
+  public PolicyChange[] policyChanges() {
+    return policyChanges != null ? policyChanges.clone() : null;
```

Comment:
Could you check the `clone()` when retrieving the policy changes from policy event?